squarelet/email_api/viewsets.py

```diff
@@ -0,0 +1,122 @@
+# Django
+from django.shortcuts import get_object_or_404
+
+# Third Party
+from allauth.account import signals
+from allauth.account.forms import AddEmailForm
+from allauth.account.models import EmailAddress, EmailConfirmation, EmailConfirmationHMAC
+from rest_framework import mixins, status, viewsets
+from rest_framework.response import Response
+
+# Squarelet
+from squarelet.email_api.serializers import PressPassEmailAddressSerializer
+
+
+class PressPassEmailConfirmationViewSet(
+    mixins.UpdateModelMixin,
+    viewsets.GenericViewSet,
+):
+    queryset = EmailConfirmation.objects.none()
+    lookup_field = "key"
+
+    def update(self, request, key=None, partial=False):
+        try:
+            confirmation = EmailConfirmationHMAC.from_key(key)
+            confirmation.confirm(self.request)
+            serializer = PressPassEmailAddressSerializer(confirmation.email_address)
+            return Response(serializer.data)
+
+        except EmailConfirmation.DoesNotExist:
+            return Response(
+                "Please specify a valid key", status=status.HTTP_400_BAD_REQUEST
+            )
+
+
+class PressPassEmailAddressViewSet(
+    mixins.CreateModelMixin,
+    mixins.UpdateModelMixin,
+    mixins.ListModelMixin,
+    mixins.DestroyModelMixin,
+    viewsets.GenericViewSet,
+):
+    queryset = EmailAddress.objects.none()
+    lookup_field = "email"
+    lookup_value_regex = "[^/]+"
+    serializer_class = PressPassEmailAddressSerializer
+
+    def get_object(self, email):
```

Comment:
Why are you overriding `get_object`?  It should return the correct email instance based on the URL instead of passing the email address in as a parameter.  I think this depends on overriding `get_queryset` to select all email addresses for the current user, so we do not allow users to interact with other user's email addresses